tests/test_config/test_readonly_containers.py

```diff
@@ -0,0 +1,116 @@
+import pytest
+
+from sacred.config.custom_containers import make_read_only, ReadOnlyList, \
+    ReadOnlyDict
+from sacred.utils import SacredError
+
+
+def _check_read_only_dict(d):
+    assert isinstance(d, ReadOnlyDict)
+
+    raises = pytest.raises(
+        SacredError, match='This ReadOnlyDict is read-only!')
+
+    if len(d) > 0:
+        # Test removal of entries and overwrite an already present entry
+        key = list(d.keys())[0]
+
+        with raises:
+            d[key] = 42
+
+        with raises:
+            del d[key]
+
+        with raises:
+            d.pop(key)
+
+    # Test direct writes
+    with raises:
+        d['abcdefg'] = 42
+
+    # Test other functions that modify the dict
+    with raises:
+        d.clear()
+
+    with raises:
+        d.update({'abcdefg': 42})
+
+    with raises:
+        d.popitem()
+
+    with raises:
+        d.setdefault(a=0)
+
+
+def _check_read_only_list(l):
+    assert isinstance(l, ReadOnlyList)
+
+    raises = pytest.raises(
```

Comment:
Maybe rename this to something different as below, to make it more explicit that it is not the default `pytest.raises` function that is used here.